src/Stream.php

```diff
@@ -40,6 +40,13 @@
  */
 class Stream implements StreamInterface
 {
+    use StreamTrait;
+
+    /**
+     * A list of types that are allowed to instantiate a Stream
+     */
+    const ALLOWED_TYPES = ['stream', 'gd'];
```

Comment:
modifier can be added, eg: private

Reply:
I don't have an objection, but would that really be necessary?

Reply:
Actually, it seems necessary to keep it public, as they're referenced in the call to `isValidStreamResourceType` in `StreamFactory`, on line 51.

Reply:
/cc @weierophinney

Reply:
After a bit of thought, @samsonasik, I moved Stream::ALLOWED_TYPES to be a private variable in StreamTrait. Thanks for the suggestion and encouragement, and thanks to @weierophinney for further educating me.